src/fswAlgorithms/attGuidance/sunSafePointCpp/sunSafePointCpp.cpp

```diff
@@ -62,23 +61,71 @@ void SunSafePointCpp::Reset(uint64_t callTime)
  @return void
  @param callTime [ns] Time the method is called
 */
-void SunSafePointCpp::UpdateState(uint64_t callTime)
-{
-    // Create the buffer messages
-    NavAttMsgPayload sunDirectionInBuffer;
-    NavAttMsgPayload localImuDataInBuffer;
-    AttGuidMsgPayload attGuidanceOutBuffer;
-
+void SunSafePointCpp::UpdateState(uint64_t callTime) {
     // Zero the attitude guidance output buffer message
-    attGuidanceOutBuffer = AttGuidMsgPayload();
+    this->attGuidanceOutBuffer = AttGuidMsgPayload();
 
     // Read the current sun body vector estimate input message
-    sunDirectionInBuffer = NavAttMsgPayload();
+    this->sunDirectionInBuffer = NavAttMsgPayload();
     if (this->sunDirectionInMsg.isWritten()) {
-        sunDirectionInBuffer = this->sunDirectionInMsg();
+        this->sunDirectionInBuffer = this->sunDirectionInMsg();
+    }
+
+    // Determine norm of measured Sun-direction vector
+    double sHatNorm = cArray2EigenVector3d(this->sunDirectionInBuffer.vehSunPntBdy).norm();
+
+    // Computing the attitude guidance states sigma_BR and omega_RN_B
+    if (this->sunDirectionIsAvailable(sHatNorm)) {
+        this->computeAttGuidanceStates(sHatNorm);
+    } else {
+        v3SetZero(this->attGuidanceOutBuffer.sigma_BR);
     }
 
+    // Compute the hub angular rate error omega_BR_B
+    this->computeHubAngularRateError();
+
+    // Write the guidance output message
+    eigenVector3d2CArray(this->omega_RN_B, this->attGuidanceOutBuffer.omega_RN_B);
+    this->attGuidanceOutMsg.write(&this->attGuidanceOutBuffer, moduleID, callTime);
+}
+
+/*! Method for computing the attitude guidance states sigma_BR and omega_RN_B if a valid sun direction vector is available.
+ @return void
+ @param sHatNorm Norm of measured Sun-direction vector
+*/
+void SunSafePointCpp::computeAttGuidanceStates(double sHatNorm) {
+    double dotProductNormalized = (this->sHatBdyCmd.dot(cArray2EigenVector3d(this->sunDirectionInBuffer.vehSunPntBdy)))
+                                          / sHatNorm;
+    dotProductNormalized = fabs(dotProductNormalized) > 1.0 ?
+    dotProductNormalized/fabs(dotProductNormalized) : dotProductNormalized;
+    this->sunAngleErr = safeAcos(dotProductNormalized);
+
+    // Compute the heading error relative to the sun direction vector
+    Eigen::Vector3d e_hat;  // Eigen Axis
```

Comment:
:large_orange_diamond: We should declare this variable within the scope that it is used. This would be inside the 'else' clause.

Reply:
Done